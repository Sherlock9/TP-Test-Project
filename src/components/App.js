import React, {Component} from 'react';
import ParseRupiah from './ParseRupiah';

class App extends Component {
  render() {
    return (
      <div className="body">
        <div className="header">
          <h1>Rupiah Parser</h1>
        </div>
        <ParseRupiah />
        <div className="footer">
          <p>By Ev Robinson</p>
        </div>
      </div>
    );
  }
}

export default App;
