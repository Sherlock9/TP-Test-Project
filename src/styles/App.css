html,
body {
  height: 100%;
  margin: 0;
  font-family: Arial, Helvetica, sans-serif;
}
body {
  display: flex;
  flex-direction: column;
}
div {
  display: flex;
  flex: 1;
  flex-direction: column;
}
.header {
  flex: 0;
  background-color: #d3dfe9;
}
h1 {
  color: #007a7c;
  text-align: center;
}
.body {
  justify-content: stretch;
}
.footer {
  background-color: #d3dfe9;
  flex: 0;
  align-items: center;
}
