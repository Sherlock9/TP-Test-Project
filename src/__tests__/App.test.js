import {shallow, mount, render} from 'enzyme';
import App from '../components/App.js';
import React from 'react';

import renderer from 'react-test-renderer';

it('App snapshot', () => {
  let tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
