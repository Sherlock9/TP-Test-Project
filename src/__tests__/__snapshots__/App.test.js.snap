// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="body"
>
  <div
    className="header"
  >
    <h1>
      Rupiah Parser
    </h1>
  </div>
  <div>
    <div
      className="input-container"
    >
      <form>
        <input
          id="rupiah"
          onChange={[Function]}
          onSubmit={[Function]}
          placeholder="Input rupiah amount"
          type="text"
          value=""
        />
        <button
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      className="bad-input"
    />
    <div
      className="results"
    >
      <table>
        <thead>
          <tr>
            <td>
              Denomination
            </td>
            <td>
              Amount
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            className="table-odd"
          >
            <td>
              100000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              50000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr
            className="table-odd"
          >
            <td>
              20000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              10000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr
            className="table-odd"
          >
            <td>
              5000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              1000
            </td>
            <td>
              0
            </td>
          </tr>
          <tr
            className="table-odd"
          >
            <td>
              500
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              200
            </td>
            <td>
              0
            </td>
          </tr>
          <tr
            className="table-odd"
          >
            <td>
              100
            </td>
            <td>
              0
            </td>
          </tr>
          <tr>
            <td>
              50
            </td>
            <td>
              0
            </td>
          </tr>
          <tr
            className="table-odd"
          >
            <td>
              Remainder
            </td>
            <td>
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="footer"
  >
    <p>
      By Ev Robinson
    </p>
  </div>
</div>
`;
